### DISCUSSION TITLE
How do you deal with the lacking access to params and URL in the client when using shadow endpoints

### DISCUSSION THREAD

**Discussion by lkj4** [sveltejs/kit#4132] [Q&A] [answered]
Pretty much the title. Do you return those as the response in the shadow endpoint just manually or is there a smarter way?

I need to access them often in the client and cannot rely on `window` because of SSR.

**Comment by mrkishi** [comment, marked as answer]
I guess it depends on what you need it for. If it's for something exclusively on the client, you could use [the `page` store](https://kit.svelte.dev/docs/modules#$app-stores-page) instead.

**Reply by lkj4** [reply]
Wow, thanks! Didn't know that such a thing exists, nice 😀

**Reply by lkj4** [reply]
hm, that doesn't seem to work on the server-side when SSRing 😶

**Reply by mustofa-id** [reply]
There is [URL](https://kit.svelte.dev/docs/types#:~:text=string%3E%3E%20%7B%0A%20%20request%3A%20Request%3B-,url,-%3A%20URL%3B%0A%20%20params%3A%20Params) object in request event in endpoint parameter that work on server-side.

**Reply by lkj4** [reply]
Thanks what I meant is that `page.url` from `$app/stores` doesn't work in the client when the client is prerendered on the server. The solution is that the endpoint sends the url in the body to the client, so it can access when the client is being prerendered on the backend.

**Reply by ignatiusmb** [reply]
Can you explain the issue more clearly? The `page` store is the way to access the info you needed in the client-side and it will always be available there. Prerendering is another topic and was not mentioned in the original issue, so marking this answer as the correct one in the meantime, and I suggest you to read more about this at https://kit.svelte.dev/docs/page-options#prerender-when-not-to-prerender.
